### PR TITLE
Update league/commonmark to 1.4.0

### PR DESCRIPTION
**Pull request by glensc**
This allows using new extensions:
- Added new Heading Permalink extension (https://github.com/thephpleague/commonmark/pull/420)
- Added new Table of Contents extension (https://github.com/thephpleague/commonmark/issues/441)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
 		"horde/text-flowed": "dev-patch-1 as 2.0.3",
 		"horde/util": "dev-patch-1 as 2.5.8",
 		"ircmaxell/random-lib": "^1.2",
-		"league/commonmark": "^1.3",
+		"league/commonmark": "^1.4",
```
Comment (GrahamCampbell) [inline composer.json:47]: Since you are using 1.4, you should change the code to use the non-deprecated classes.

Comment (glensc) [reply composer.json:47]: @GrahamCampbell thanks for noticing.

This is still WIP (GitHub doesn't allow on/off WIP state, like GitLab and I forgot on initial PR creation)

Comment (GrahamCampbell) [reply composer.json:47]: > GitHub doesn't allow on/off WIP state

It does actually. They launched this a couple of weeks ago. ;)

Comment (glensc) [reply composer.json:47]: @GrahamCampbell: oh.

**Pull request merged**